Helper.java

```diff
@@ -16,7 +14,11 @@ static void enterData() throws InvalidAnimalName {
         int age;
         boolean isReproductive;
         boolean repeat = true;
-        List<Object> animals = new ArrayList<>();
+        List<Animal> animals = new ArrayList<>();
```

Comment:
You are creating additional list for collecting all items. It is dublicating
Basically that is not good in terms of maintanance. In case you need to change an object - you need not to forget to do it twice